### PR TITLE
Add second programming test

### PR DESCRIPTION
**Pull request by michaelwebb76**

### CODE REVIEW

**Comment by cdaloisio** [comment]
LGTM 👍

File: README.md [hunk 1]
```diff
@@ -36,8 +67,6 @@ Conditions:
 1. You must do this in a language that you have not programmed in before.
 2. You must not use a library that provides a `reverse` method or function.
 
-That's it!
-
 Here are some of the languages (in no particular order) that we use at Trike on the side to help us
 solve problems.
 [Rust](https://www.ruby-lang.org/en/), [Swift](https://swift.org), [Haskell](haskell.org),
```
Comment (joycse06) [inline README.md:80]: `[Rust](https://www.rust-lang.org/en-US/)` ?

File: README.md [hunk 2]
```diff
@@ -17,7 +23,32 @@ string and see what result you get (hint: you'll know when you've done it right)
...
+honour [BODMAS](https://www.mathsisfun.com/operation-order-bodmas.html) but you don't need to worry
+about the BO, just the DMAS.
+
+For extra points, do something clever that reduces the average number of guesses your class needs
+to make for each call to `guess`.
+
+e.g.
+
+```
+> equation_guesser.new([1, 2, 3, 4], 24)
```
Comment (joycse06) [inline README.md]: `Equation_guesser.new([1, 2, 3, 4], 24)` ?

File: README.md [hunk 3]
```diff
@@ -17,7 +25,38 @@ string and see what result you get (hint: you'll know when you've done it right)
 !YTIRCO!IQIIQIDEMGMMIM FO YMJMMSM!RA !EGEEJEHT ROEOOSOF PAEJEEBEL TN!AIKIITIG ENVNNMNO ,GQGGCGN!ILEKIZIISIRT A RJRRDROF PETOTTJTS LLZLLEL!AMSXSSMS ENODOOSO
 ```
 
-## Expanding your programming comprehension
+## Problem 2 - Number Wrangling
+
+Write a `EquationGuesser` class that in its constructor takes as one argument an array of integers
```
Comment (jack-trikeapps) [inline README.md:30]: Again, presupposes OO.

Comment (jack-trikeapps) [reply README.md:30]: also, consider "sequence" as a more generic term than "array"

File: README.md [hunk 4]
```diff
@@ -17,7 +25,38 @@ string and see what result you get (hint: you'll know when you've done it right)
...
+(we'll call these the operands), and as another argument another integer (we'll call this the
+desired result). Your class should provide a method `guess` that takes an integer (guesses).
+When `guess` is called, your class should attempt to output `guesses` X string representations of
```
Comment (jack-trikeapps) [inline README.md:33]: It isn't really clear that you're trying to determine the operators that make the equation meet the guess.

File: README.md [hunk 5]
```diff
@@ -17,7 +25,38 @@ string and see what result you get (hint: you'll know when you've done it right)
 !YTIRCO!IQIIQIDEMGMMIM FO YMJMMSM!RA !EGEEJEHT ROEOOSOF PAEJEEBEL TN!AIKIITIG ENVNNMNO ,GQGGCGN!ILEKIZIISIRT A RJRRDROF PETOTTJTS LLZLLEL!AMSXSSMS ENODOOSO
 ```
 
-## Expanding your programming comprehension
+## Problem 2 - Number Wrangling
+
+Write a `EquationGuesser` class that in its constructor takes as one argument an array of integers
+(we'll call these the operands), and as another argument another integer (we'll call this the
+desired result). Your class should provide a method `guess` that takes an integer (guesses).
```
Comment (jack-trikeapps) [inline README.md:32]: quote `guesses`

File: README.md [hunk 6]
```diff
@@ -17,7 +25,38 @@ string and see what result you get (hint: you'll know when you've done it right)
...
+> equation_guesser = EquationGuesser.new([1, 2, 3, 4], 24)
+> equation_guesser.guesses(3)
+1 + 2 + 3 + 4 = 10 (-14)
+1 * 2 * 3 * 4 = 24 (0)
+Guesses: 2 Best Guess: 1 * 2 * 3 * 4 = 24 (0)
+```
+
+If the equation is exactly equal to the desired result, the program should exit. For the sake of
+this exercise, you only need consider positive integers (i.e. use integer division). Your equations
+should honour [BODMAS](https://www.mathsisfun.com/operation-order-bodmas.html) but you don't need to
```
Comment (jack-trikeapps) [inline README.md:49]: How about "standard operator precedence" or "the usual rules of arithmetic", with a side of "don't worry about bracketing subexpressions when generating guesses"?

File: README.md (hunk 4 above)
Comment (jack-trikeapps) [reply README.md:33]: I think to make the requirements clearer, I'd start with a description of the problem (guessing the operators between integers to make the equation true), then put the implementation skeleton after that.

**Review by jack-trikeapps — COMMENTED** [review]
hi i am jack and i have opinions

File: README.md (hunk 1 above)
Comment (jack-trikeapps) [reply README.md:80]: Underhanded rubyist propaganda, I tells ya!

Also "haskell.org" should be a proper URL.

**Pull request merged**